cmd/psa/common.go

```diff
@@ -23,31 +23,32 @@ func loadTokenFromFile(fs afero.Fs, fn string) (*psatoken.Evidence, error) {
 	return e, nil
 }
 
-func loadClaimsFromFile(fs afero.Fs, fn string, partial bool) (psatoken.IClaims, error) {
+func loadClaimsFromFile(fs afero.Fs, fn string, validate bool) (psatoken.IClaims, error) {
 	buf, err := afero.ReadFile(fs, fn)
 	if err != nil {
 		return nil, err
 	}
 
-	// Check first for P2 Claims
+	return claimsFromJSON(buf, validate)
+}
+
+func claimsFromJSON(j []byte, validate bool) (psatoken.IClaims, error) {
 	p2 := &psatoken.P2Claims{}
 
-	err2 := json.Unmarshal(buf, p2)
+	err2 := json.Unmarshal(j, p2)
 	if err2 == nil {
-		err2 = p2.Validate()
-		if partial || (err2 == nil) {
+		if validate || p2.Validate() == nil {
```

Comment:
this don't look right: imagine p1 decoding+validation fails too, then the error message will report `"... and p2 error: (nil)"`

BTW this condition should be easy to test by supplying an empty JSON

Reply:
ISTM that you want something like:

```
if err2 == nil {
        if validate {
                err2 = p2.Validate()
                if err2 == nil {
                        return p2, nil
                }
        }
}

// continue with P1
```

Reply:
Done and verified to be working!